### ISSUE TITLE
when errors occur on a watch, the watch is dropped without being recreated

### ISSUE THREAD

**Issue by PeterGrace** [PeterGrace/pull-secret-operator#3]
I need to implement a re-initialize type functionality to reestablish the watches when there's an error detected.

**Comment by PeterGrace** [comment]
I believe I've solved this with 4a32e61.

**Comment by PeterGrace** [comment]
Yep, this is now fixed -- the error reoccured overnight and the watch successfully restarted:

```
pull-secret-operator-867684f564-8jkcr pull-secret-operator [2020-10-23T08:31:35Z INFO  pull_secret_operator::crd_watcher] retry 0:Received error: error returned by apiserver during watch: too
 old resource version: 66287209 (66287210): Expired
pull-secret-operator-867684f564-8jkcr pull-secret-operator [2020-10-23T08:31:45Z INFO  pull_secret_operator::crd_watcher] restarted: dreg-registry
pull-secret-operator-867684f564-8jkcr pull-secret-operator [2020-10-23T08:31:45Z INFO  pull_secret_operator::crd_watcher] Preparing to copy jenkins/dreg-registry to namespace mayan
pull-secret-operator-867684f564-8jkcr pull-secret-operator [2020-10-23T08:31:45Z INFO  pull_secret_operator::crd_watcher] Secret mayan/dreg-registry already exists, updating.
pull-secret-operator-867684f564-8jkcr pull-secret-operator [2020-10-23T08:31:45Z INFO  pull_secret_operator::crd_watcher] Preparing to copy jenkins/dreg-registry to namespace nginx-ingress
pull-secret-operator-867684f564-8jkcr pull-secret-operator [2020-10-23T08:31:45Z INFO  pull_secret_operator::crd_watcher] Secret nginx-ingress/dreg-registry already exists, updating.
pull-secret-operator-867684f564-8jkcr pull-secret-operator [2020-10-23T08:31:45Z INFO  pull_secret_operator::crd_watcher] Preparing to copy jenkins/dreg-registry to namespace monitoring
pull-secret-operator-867684f564-8jkcr pull-secret-operator [2020-10-23T08:31:45Z INFO  pull_secret_operator::crd_watcher] Secret monitoring/dreg-registry already exists, updating.
pull-secret-operator-867684f564-8jkcr pull-secret-operator [2020-10-23T08:31:45Z INFO  pull_secret_operator::crd_watcher] Preparing to copy jenkins/dreg-registry to namespace faas-functions
pull-secret-operator-867684f564-8jkcr pull-secret-operator [2020-10-23T08:31:45Z INFO  pull_secret_operator::crd_watcher] Secret faas-functions/dreg-registry already exists, updating.
```
